src/components/intent.ts

```diff
@@ -806,13 +806,16 @@ export class Intent {
         // eslint-disable-next-line camelcase
         room_id: string
     }) {
-        if (!this._membershipStates || !this._powerLevels) {
-            return;
-        }
         if (event.state_key === undefined) {
             // We MUST operate on state events exclusively
             return;
         }
+        // Invalidate the state cache if anything changes in the state.
+        this._requestCaches.roomstate.invalidate(event.room_id);
```

Comment:
Isn't there a condition here where we call `invalidate()` then immediately return without doing anything?

Reply:
Yup, but this cache basically caches calls to intent.getRoomState. Basically every time we see a state event happen in a room, we want to invalidate that cache. Otherwise, we don't really care if this function doesn't do anything.